### ISSUE TITLE
Fine grained access tokens

### ISSUE THREAD

**Issue by BaseCrusher** [go-gitea/gitea#20984]
### Feature Description

I believe it would be a good feature if users could fine grain their access tokens. This would make it possible to limit what is possible with a specific token and introduce an additional layer of security. 
Especially now that we have more features within gitea than just git management this would be a very neet addition.

**Comment by KN4CK3R** [comment]
Duplicate of #8960

**Comment by BaseCrusher** [comment]
Yes but I would suggest to either make the other issue more specific or to keep this one. 
Thinks like what packages registries can a token be used for and what action can be used with an token. So it's not only for read only access.

If you think this is being stated in  https://github.com/go-gitea/gitea/issues/8960 then feel free  to close this issue.

**Comment by KN4CK3R** [comment]
No everything fine. Just wanted to link the other issue.

**Comment by TheDom42** [comment]
Just for clarification, as I might have run into an issue here: I have configured a mirror repo to pull clone a github repo in my account, using the "new"/beta fine-grained access token (instead of classical tokens).

When setting up the repo in gitea, everything works perfectly and the inital pull is fast. However, the automatic cron pull as well as the manual pull fail complete silently:
1. The user initiating the pull manually receives the message that the repos will be synced
2. The cron is marked as "last run at..."
However, in both cases, no sync actually takes place!

As I'm rather new to gitea, I finally stumbled upon the admin logs which are spammed with messages looking like this:
```
Failed to update mirror repository '<REDACTED>': fatal: could not read Username for 'https://github.com': 
terminal prompts disabled
error: could not fetch origin
```

I have set tup the personal access tokens with read-only permissions and only allow access to that specific repo on Github. As the inital setup with the token worked, I was very surprised that the mirror updates fail afterwards.
Just as clarification for me: are the new Github "fine grained access tokens" fully supported yet? If so, which of the repo access rights (https://docs.github.com/en/rest/overview/permissions-required-for-fine-grained-personal-access-tokens?apiVersion=2022-11-28) are mandatory to work for a read-only pull mirror? Do I actually need to set some of them to write permissions (which I would like to avoid)?

**Comment by KN4CK3R** [comment]
@TheDom42 What you are describing is totally different than what this issue is about.

**Comment by TheDom42** [comment]
Sorry, I must have completely misinterpreted the term "Fine grained access token". I thought this was specifically referring to the Github feature and thought that this issue was maybe about the fact that they are not yet supported by Gitea. I will move the described problem to a new one.

**Comment by jolheiser** [comment]
Resolved by #20908
